pkg/registry/service/portallocator/controller/repair.go

```diff
@@ -86,10 +86,8 @@ func (c *Repair) runOnce() error {
 	if err != nil {
 		return fmt.Errorf("unable to refresh the port block: %v", err)
 	}
-
 	ctx := api.WithNamespace(api.NewDefaultContext(), api.NamespaceAll)
-	options := &api.ListOptions{ResourceVersion: latest.ObjectMeta.ResourceVersion}
-	list, err := c.registry.ListServices(ctx, options)
+	list, err := c.registry.ListServices(ctx, nil)
```

Comment:
Ah, this code doesn't happen on startup, right?-- a write probably eventually kicked it out of the wait loop. So the locking bug probably broke this. It used to be papered over by lots of churn in etcd.